### PR TITLE
[MXNet] Workaround copyTo operator that produces corrupted NDArray in multithreading environment

### PR DESCRIPTION
**Pull request by stu1130**
… multithreading environment

## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: api/src/main/java/ai/djl/training/ParameterStore.java [hunk 1]
```diff
@@ -119,14 +119,12 @@ public NDArray getValue(Parameter parameter, Device device, boolean training) {
...
                 if (copy || !array.getDevice().equals(device)) {
-                    synchronized (array) {
-                        array = array.toDevice(device, true);
-                        array.attach(manager);
-                        // some parameter doesn't require grad
-                        // for example running_mean in BatchNorm
-                        if (parameter.requireGradient() && training) {
-                            array.attachGradient();
-                        }
+                    array = array.toDevice(device, true);
```
Comment (lanking520) [inline api/src/main/java/ai/djl/training/ParameterStore.java:122]: We still need to evaluate this change. Can you ask them to test again if you would like to apply this fix?

Comment (stu1130) [reply api/src/main/java/ai/djl/training/ParameterStore.java:122]: I am running testing against this change

Comment (stu1130) [reply api/src/main/java/ai/djl/training/ParameterStore.java:122]: It turns out without adding the lock here, we still observe the SparseFormat error

**Comment by lanking520** [comment]
Do we want to close this?

**Pull request closed without merging**